integration/spark-common/src/main/java/org/apache/carbondata/spark/load/CarbonLoaderUtil.java

```diff
@@ -258,92 +258,105 @@ public static void deleteLocalDataLoadFolderLocation(CarbonLoadModel loadModel,
    * @throws IOException
    */
   public static boolean recordLoadMetadata(LoadMetadataDetails newMetaEntry,
-      CarbonLoadModel loadModel, boolean loadStartEntry, boolean insertOverwrite)
+      CarbonLoadModel loadModel, boolean loadStartEntry, boolean insertOverwrite, boolean isUpdate)
```

Comment:
I think we may not require to change this code, In update case this method shouldn't be called

Reply:
Because should set carbonLoadModel in recordLoadMetadata(), otherwise the below code in carbonTableSchema.scala will has NLP issue: 
if (carbonLoadModel.getLoadMetadataDetails.isEmpty && carbonLoadModel.getUseOnePass &&
              StringUtils.isEmpty(column_dict) && StringUtils.isEmpty(all_dictionary_path)) {
            LOGGER.info(s"Cannot use single_pass=true for $dbName.$tableName during the first load")
            LOGGER.audit(s"Cannot use single_pass=true for $dbName.$tableName during the first load")
            carbonLoadModel.setUseOnePass(false)
          }

in carbonTableSchema.scala , the below code need to